USER_MANUAL.md

```diff
@@ -3042,6 +3042,85 @@ Using `rbreak <regex>` you can set breakpoints on multiple functions.
+Once this is done, you then build K with optimizations and debug info enabled, like so: 
+```shell
+mvn package -Dproject.build.type="RelWithDebInfo"
+```
+
+3. Next, you build the semantics with optimizations and debug info enabled (i.e., `kompile -O3 -ccopt -g` )
+
+4. Once all this is done, you should be ready to profile your application. Essentially, you should run whatever test suite you usually run, but with `perf record -g -- ` prefixed to the front. For example, for KEVM it's the following command.
+```shell
+perf record -g -- make test-conformance
+```
+For best data, don't run this step in parallel.
+
+4. Finally, you want to filter out just the samples that landed within the llvm backend and view the report. For this, you need to know the name of the binary that was generated by your build system. Normally it is `interpreter`, but e.g. if you are building the web3 client for kevm, it would be `kevm-client`. You will want to run `perf report -g -c $binary_name`. If all goes well, you should see a breakdown of where CPU time has been spent executing the application. You will know that sufficient time was spent rebuilding dependencies with the correct flags when the total time reported by the main method is close to 100%. If it's not close to 100%, this is probably because a decent amount of self time was reported in stack traces that were not built with frame pointers enabled, meaning that perf was unable to walk the stack. You will have to go back, rebuild the appropriate libraries, and then record your trace again.
+
+5. Your ultimate goal is to identify the hotspots that take the most time, and make them execute faster. Entries like `step` and `step_1234` like functions refer to the cost of matching. An entry like `side_condition_1234` is a side condition and `apply_rule_1234` is constructing the rhs of a rule. You can convert from this rule ordinal to a location using the `llvm-kompile-compute-loc` script in the bin folder of the llvm backend repo. For example, 
+```shell
+llvm-kompile-compute-loc 5868 evm-semantics/.build/defn/llvm/driver-kompiled
+```
+spits out the following text.
+```
+Line: 18529
+/home/dwightguth/evm-semantics/./.build/defn/llvm/driver.k:493:10
+```
+This is the line of `definition.kore` that the axiom appears on as well as the original location of the rule in the K semantics. You can use this information to figure out which rules and functions are causing the most time and optimize them to be more efficient.
+
+## Makefile target examples
```

Comment:
I think this section just makes the above more confusing and should probably be removed.